pkg/client/unversioned/conditions.go

```diff
@@ -41,3 +42,26 @@ func ControllerHasDesiredReplicas(c Interface, controller *api.ReplicationContro
 		return ctrl.Status.ObservedGeneration >= desiredGeneration && ctrl.Status.Replicas == ctrl.Spec.Replicas, nil
 	}
 }
+
+// JobHasDesiredParallelism returns a condition that will be true if the desired parallelism count
+// for a job equals the current active counts or is less by an appropriate successful/unsuccessful count.
+func JobHasDesiredParallelism(c Interface, job *experimental.Job) wait.ConditionFunc {
+
+	return func() (bool, error) {
+		job, err := c.Experimental().Jobs(job.Namespace).Get(job.Name)
+		if err != nil {
+			return false, err
+		}
+
+		// desired parallelism can be either the exact number, in which case return immediately
+		if job.Status.Active == *job.Spec.Parallelism {
+			return true, nil
+		}
+		// otherwise count successful and unsuccessful as well
+		progress := *job.Spec.Completions - job.Status.Active - job.Status.Successful
+		if job.Spec.Template.Spec.RestartPolicy == api.RestartPolicyNever {
+			progress -= job.Status.Unsuccessful
```

Comment:
This should be removed based on #14142 -- specifically e29e606792c69d6aa03f31aee7b5669799dead19.